modules/ballerina-http/src/main/java/org/ballerinalang/net/http/actions/AbstractHTTPAction.java

```diff
@@ -54,66 +54,111 @@
 
     private static final Logger logger = LoggerFactory.getLogger(AbstractHTTPAction.class);
 
-    private static final String BALLERINA_USER_AGENT;
+    private static final String CACHE_BALLERINA_VERSION;
+    static {
+        CACHE_BALLERINA_VERSION = System.getProperty(BALLERINA_VERSION);
+    }
 
-    /* Application level timeout */
-    private static final long SENDER_TIMEOUT = 180000; // TODO: Make this configurable with endpoint timeout impl
+    protected HTTPCarbonMessage createCarbonMsg(Context context) {
 
-    static {
-        String version = System.getProperty(BALLERINA_VERSION);
-        if (version != null) {
-            BALLERINA_USER_AGENT = "ballerina/" + version;
-        } else {
-            BALLERINA_USER_AGENT = "ballerina";
-        }
+        // Extract Argument values
+        BConnector bConnector = (BConnector) getRefArgument(context, 0);
+        String path = getStringArgument(context, 0);
+        BStruct requestStruct  = ((BStruct) getRefArgument(context, 1));
+        //TODO check below line
```

Comment:
Can we remove this TODO? If not shall we use a proper message for future reference